internal/graphql/graph/schema.resolvers.go

```diff
@@ -0,0 +1,118 @@
+package graph
+
+// This file will be automatically regenerated based on the schema, any resolver implementations
+// will be copied through when generating and any unknown code will be moved to the end.
+// Code generated by github.com/99designs/gqlgen version v0.17.27
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kubeshop/testkube/internal/graphql/graph/model"
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+	executorsmapper "github.com/kubeshop/testkube/pkg/mapper/executors"
+	"github.com/kubeshop/testkube/pkg/rand"
+)
+
+// Type is the resolver for the type field.
+func (r *eventResolver) Type(ctx context.Context, obj *testkube.Event) (*string, error) {
+	panic(fmt.Errorf("not implemented: Type - type"))
+}
+
+// Resource is the resolver for the resource field.
+func (r *eventResolver) Resource(ctx context.Context, obj *testkube.Event) (*string, error) {
+	panic(fmt.Errorf("not implemented: Resource - resource"))
+}
+
+// Features is the resolver for the features field.
+func (r *executorResolver) Features(ctx context.Context, obj *testkube.Executor) ([]*model.FeaturesListItem, error) {
+	panic(fmt.Errorf("not implemented: Features - features"))
+}
+
+// Labels is the resolver for the labels field.
+func (r *executorResolver) Labels(ctx context.Context, obj *testkube.Executor) (*string, error) {
+	panic(fmt.Errorf("not implemented: Labels - labels"))
+}
+
+// Event is the resolver for the event field.
+func (r *executorDetailsResolver) Event(ctx context.Context, obj *testkube.ExecutorDetails) (*testkube.Event, error) {
+	panic(fmt.Errorf("not implemented: Event - event"))
+}
+
+// Tooltips is the resolver for the tooltips field.
+func (r *executorMetaResolver) Tooltips(ctx context.Context, obj *testkube.ExecutorMeta) (*string, error) {
+	panic(fmt.Errorf("not implemented: Tooltips - tooltips"))
+}
+
+// Executors is the resolver for the executors field.
+func (r *queryResolver) Executors(ctx context.Context) ([]*testkube.Executor, error) {
+	panic(fmt.Errorf("not implemented: Executors - executors"))
+}
+
+// Executors is the resolver for the executors field.
+func (r *subscriptionResolver) Executors(ctx context.Context) (<-chan []testkube.ExecutorDetails, error) {
+	ch := make(chan []testkube.ExecutorDetails)
+
+	// TODO You can (and probably should) handle your channels in a central place outside of `schema.resolvers.go`.
+	// For this example we'll simply use a Goroutine with a simple loop.
+	go func() {
+		r.Log.Infof("%+v\n", "subscribed to events.executor.>")
+
+		r.Bus.SubscribeTopic("events.executor.>", rand.String(30), func(e testkube.Event) error {
```

Comment:
Is this how it should be used?

Reply:
**TLDR:** it is often used this way and it's rather perfectly suited for our needs.

The most natural choice for subscriptions is to send events (like `onExecutorsListChange` in this case). On the opposite, live queries are just regular queries, but with real-time (more/less) updates.

Live queries are not a part of GraphQL specification and are implemented in different ways. The most distinctive way is to add `@live` directive to the query, but it's not necessarily supported (i.e. `gqlgen` [**doesn't support it**](https://github.com/99designs/gqlgen/issues/2552)). Because of all the problems, a lot of tools, like [**Hasura**](https://hasura.io/docs/latest/subscriptions/postgres/livequery/index/) are using subscriptions this way (so basically replace `query` with `subscription` to obtain data).

There is [**nice article**](https://the-guild.dev/blog/subscriptions-and-live-queries-real-time-with-graphql) about this topic.

As an alternative solution, you can `query` the data before, and add `subscription` later for changes. This solution has multiple problems though:
* it's more complex
* you need to handle data from the time between the query and subscription started (i.e. querying again or providing the last update date for the subscription as param)